pkg/apis/samples/v1/types.go

```diff
@@ -440,9 +427,7 @@ func (s *Config) ClusterNeedsCreds() bool {
 		s.Spec.ManagementState == operatorv1.Unmanaged {
 		return false
 	}
-	return s.Spec.InstallType == RHELSamplesDistribution &&
-		s.ConditionFalse(ImportCredentialsExist) &&
-		(s.Spec.SamplesRegistry == "" || s.Spec.SamplesRegistry == "registry.redhat.io")
+	return s.ConditionFalse(ImportCredentialsExist) && (s.Spec.SamplesRegistry == "" || s.Spec.SamplesRegistry == "registry.redhat.io")
```

Comment:
Since we now always use the installer pullsecret as the initial creds, i think it's worth revisiting whether we get any value out of the "ImportCredentialsExist" condition anymore?

In addition, it was important to have that condition when we didn't retry imports because if we created the imagestream before the creds existed, the import would fail and never get retried once the creds did exist. Now that we have retries for image imports, it's less critical that we get the ordering right.  (If anything we should just rerun the imports when we update the creds)

Doesn't have to be in this PR, but (assuming you agree) i think it would allow us to remove a lot of complexity from the operator.

Reply:
ImportCredExists provides value if there are problems copying the secret, and facilitates retry of that copy if it failed previously... the recent clean up PR for the kube-system changes brought that element in 

Yes, the motivation for it has changed ... but I'm still inclined to say it has value, and thus we should keep it.

Reply:
I can see the value in using it to indicate issues copying the secret.

I don't agree with using it to facilitate retrying the copy.  That should be handled by the standard relist/resync logic as well as requeueing events.  (ie if we hit an error copying it, we requeue the watch event, and every 10 mins we relist the kube-system secrets and update the openshift secret with its content if it's different).

For the bootstrap itself (where there is no watch event to requeue), doing a few retries on the copy is reasonable, after which if it still fails we'd set the credsexist=false w/ error message and hope that during a resync/relist we succeed in copying.

Reply:
i'm fine with removing the extra copy attempt on the config watch path if you want, as well as adding a few retries on bootsrap

Separate PR still though, right?

Reply:
yeah, separate PR.